### DISCUSSION TITLE
How to find out, if a class member has  new-modifier by using IMethodSymbol or IPropertySymbol or IFieldSymbol

### DISCUSSION THREAD

**Discussion by Stoffelche** [dotnet/roslyn#66460] [Q&A] [answered]
Is there a way to obtain the information, wether a member of a type is declared as new from one of the symbol-interfaces mentioned in the title?

**Comment by CyrusNajmabadi** [comment, marked as answer]
Yes, if this is a symbol from source. From the ISymbol you can get back to the DeclaringSyntaxReference. From that you can find the modifiers and see if there is a `new` modifier. 

You cannot tell from metadata as this information is not stored there. It's purely a source-only construct used to suppress a warning.

**Reply by Stoffelche** [reply]
Thanks
